common/src/main/java/tachyon/security/TachyonLoginModule.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.security;
+
+import java.security.Principal;
+import java.util.Map;
+import java.util.Set;
+
+import javax.security.auth.Subject;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.spi.LoginModule;
+
+/**
+ * A login module that search the Kerberos or OS user from Subject, and then convert to a Tachyon
+ * user. It does not really authenticate the user in its login method.
+ */
+public class TachyonLoginModule implements LoginModule {
+  private Subject mSubject;
+  private User mUser;
+
+  @Override
+  public void initialize(Subject subject, CallbackHandler callbackHandler,
+      Map<String, ?> sharedState, Map<String, ?> options) {
+    mSubject = subject;
+  }
+
+  /**
+   * Authenticate the user (first phase).
+   *
+   * The implementation does not really authenticate the user here. Always return true.
+   * @return true in all cases
+   * @throws javax.security.auth.login.LoginException
+   */
+  @Override
+  public boolean login() throws LoginException {
+    return true;
+  }
+
+  /**
+   * Abort the authentication (second phase).
+   *
+   * This method is called if the LoginContext's overall authentication failed. (login failed)
+   * Since we do nothing in login method and no internal states need to be cleaned,
+   * the implementation here is just returning true.
+   * @return true in all cases
+   * @throws LoginException
+   */
+  @Override
+  public boolean abort() throws LoginException {
+    return true;
```

Comment:
It is technically possible that mUser is assigned during `commit` but `commit` fails, so maybe still need some cleaning?